api/src/main/java/ca/bc/gov/educ/graddatacollection/api/rules/demographic/ruleset/V118DemographicStudentStatus.java

```diff
@@ -0,0 +1,71 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  | ID   | Severity | Rule                                                                  | Dependent On |
+ *  |------|----------|-----------------------------------------------------------------------|--------------|
+ *  | V118 | ERROR    |  Student Status must match PEN student status                         |  v117        |
+ *  |      |          |                                     	                              |              |
+ *
+ */
+
+@Component
+@Slf4j
+@Order(1800)
+public class V118DemographicStudentStatus implements DemographicValidationBaseRule {
+
+    private final DemographicRulesService demographicRulesService;
+
+    public V118DemographicStudentStatus(DemographicRulesService demographicRulesService) {
+        this.demographicRulesService = demographicRulesService;
+    }
+
+    @Override
+    public boolean shouldExecute(StudentRuleData studentRuleData, List<DemographicStudentValidationIssue> validationErrorsMap) {
+        log.debug("In shouldExecute of StudentStatus-V118: for demographicStudentID :: {}", studentRuleData.getDemographicStudentEntity().getDemographicStudentID());
+
+        var shouldExecute =  isValidationDependencyResolved("V18", validationErrorsMap);
+
+        log.debug("In shouldExecute of StudentStatus-V118: Condition returned - {} for demographicStudentID :: {}" ,
+                shouldExecute,
+                studentRuleData.getDemographicStudentEntity().getDemographicStudentID());
+
+        return  shouldExecute;
+    }
+
+    @Override
+    public List<DemographicStudentValidationIssue> executeValidation(StudentRuleData studentRuleData) {
+        var demStudent = studentRuleData.getDemographicStudentEntity();
+        log.debug("In executeValidation of StudentStatus-V118 for demographicStudentID :: {}", demStudent.getDemographicStudentID());
+        final List<DemographicStudentValidationIssue> errors = new ArrayList<>();
+        try {
+            var student = demographicRulesService.getStudentApiStudent(demStudent.getPen());
+            if (
+                !(
+                    demStudent.getStudentStatusCode().equalsIgnoreCase(student.getStatusCode()) ||
+                    ("A".equalsIgnoreCase(student.getStatusCode()) && "T".equalsIgnoreCase(demStudent.getStudentStatusCode()))
+                )
+            ) {
+                log.debug("StudentStatus-V118: Student Status must match PEN.  demographicStudentID :: {}", demStudent.getDemographicStudentID());
+                errors.add(createValidationIssue(StudentValidationIssueSeverityCode.ERROR, DemographicStudentValidationFieldCode.STUDENT_STATUS, DemographicStudentValidationIssueTypeCode.STUDENT_STATUS_PEN_MISMATCH));
+            }
+        } catch (GradDataCollectionAPIRuntimeException e) {
+            log.debug("StudentStatus-V118:Student No student api student record found for demographicStudentID: {}", demStudent.getDemographicStudentID());
```

Comment:
We'll need to rethrow the exception so the validation fails and retries later

Reply:
remove try catch so original throw goes through